new/components/rabbitmq/scripts/create.sh

```diff
@@ -45,26 +50,96 @@ EOF
 sudo systemctl daemon-reload
 
+ctx logger info "Chowning RabbitMQ logs path..."
+sudo chown rabbitmq:rabbitmq ${RABBITMQ_LOG_BASE}
+
 ctx logger info "Starting RabbitMQ Server in Daemonized mode..."
 sudo systemctl start cloudify-rabbitmq.service
 
 ctx logger info "Enabling RabbitMQ Plugins..."
-sudo rabbitmq-plugins enable rabbitmq_management >/dev/null
-sudo rabbitmq-plugins enable rabbitmq_tracing >/dev/null
-
-
-# enable guest user access where cluster not on localhost
-ctx logger info "Enabling RabbitMQ user access..."
-echo "[{rabbit, [{loopback_users, []}]}]." | sudo tee --append /etc/rabbitmq/rabbitmq.config >/dev/null
-
-ctx logger info "Chowning RabbitMQ logs path..."
-sudo chown rabbitmq:rabbitmq ${RABBITMQ_LOG_BASE}
+run_command_with_retries "sudo rabbitmq-plugins enable rabbitmq_management"
+run_command_with_retries "sudo rabbitmq-plugins enable rabbitmq_tracing"
+
+ctx logger info "Disabling RabbitMQ guest user"
+run_command_with_retries "sudo rabbitmqctl clear_permissions guest"
+run_command_with_retries "sudo rabbitmqctl delete_user guest"
+
+ctx logger info "Creating new RabbitMQ user and setting permissions"
+run_command_with_retries sudo rabbitmqctl add_user ${RABBITMQ_USERNAME} ${RABBITMQ_PASSWORD}
+run_noglob_command_with_retries sudo rabbitmqctl set_permissions ${RABBITMQ_USERNAME} '.*' '.*' '.*'
+
+# Deploy certificates (if applicable)
+# TODO: Validate that cert is in PEM format based on openssl command?
+if [[ "${RABBITMQ_CERT_PRIVATE}" =~ "BEGIN RSA PRIVATE KEY" ]]; then
+  ctx logger info "Found private certificate for rabbitmq."
+  echo "${RABBITMQ_CERT_PRIVATE}" | sudo tee /etc/rabbitmq/rabbit-priv.pem >/dev/null
+  # Only allow owner to see rabbit private cert
+  sudo chmod 400 /etc/rabbitmq/rabbit-priv.pem
+  sudo chown rabbitmq. /etc/rabbitmq/rabbit-priv.pem
+else
+  if [[ -z "${RABBITMQ_CERT_PRIVATE}" ]]; then
+    ctx logger info "No private certificate found."
+  else
+    ctx logger warn "Private certificate did not appear to be in PEM format."
+  fi
+fi
+# TODO: Validate that cert is in PEM format based on openssl command?
+if [[ "${RABBITMQ_CERT_PUBLIC}" =~ "BEGIN CERTIFICATE" ]]; then
+  ctx logger info "Found public certificate for rabbitmq."
+  echo "${RABBITMQ_CERT_PUBLIC}" | sudo tee /etc/rabbitmq/rabbit-pub.pem >/dev/null
+  # Allow other users to see public cert
+  sudo chmod 444 /etc/rabbitmq/rabbit-pub.pem
+  sudo chown rabbitmq. /etc/rabbitmq/rabbit-pub.pem
+else
+  if [[ -z "${RABBITMQ_CERT_PUBLIC}" ]]; then
+    ctx logger info "No public certificate found."
+  else
+    ctx logger warn "Public certificate did not appear to be in PEM format."
+  fi
+  if [[ -f /etc/rabbitmq/rabbit-priv.pem ]]; then
+    ctx logger error "Private certificate was provided but public certificate was invalid."
+  fi
+fi
+
+# Configure rabbit
+if [[ -f /etc/rabbitmq/rabbit-priv.pem ]] && [[ -f /etc/rabbitmq/rabbit-pub.pem ]]; then
+  echo "[
```

Comment:
It probably wants commenting, and likely the certificate deployment wants turning into functions so that logic to check that they're actually being created as PEM formatted SSL certs exists. Otherwise, the only easier way would actually be to remove any checks I think?